--- lopster-1.0.1/src/connection.c.orig	Sun Feb 23 11:32:30 2003
+++ lopster-1.0.1/src/connection.c	Sun Feb 23 11:33:28 2003
@@ -1265,8 +1265,14 @@
       return;
     }
   }
+  {
+      struct protoent *prt = getprotobyname("IP");
+      if (prt)
+         protocol = prt->p_proto;
+      else
+         protocol = 0;
+  }
 
-  protocol = getprotobyname("IP")->p_proto;
   sock = socket(AF_INET, SOCK_STREAM, protocol);
 
   if (sock < 0) {
